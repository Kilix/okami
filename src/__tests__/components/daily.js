import React from 'react'
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import frLocale from 'date-fns/locale/fr'
import format from 'date-fns/format'
import lolex from 'lolex'

import DailyCalendar from '../../components/daily'

describe('DailyCalendar', () => {
  beforeEach(() => {
    lolex.createClock(new Date(2017, 9, 9))
  })
  const events = [
    {
      allDay: false,
      start: new Date(2017, 9, 7, 3, 0, 0),
      end: new Date(2017, 9, 7, 14, 0, 0),
    },
    {
      allDay: false,
      start: new Date(2017, 9, 7, 5, 0, 0),
      end: new Date(2017, 9, 7, 7, 0, 0),
    },
    {
      allDay: false,
      start: new Date(2017, 9, 7, 12, 30, 0),
      end: new Date(2017, 9, 7, 13, 30, 0),
    },
    {
      allDay: false,
      start: new Date(2017, 9, 7, 13, 0, 0),
      end: new Date(2017, 9, 7, 14, 30, 0),
    },
    {
      allDay: false,
      start: new Date(2017, 9, 7, 16, 30, 0),
      end: new Date(2017, 9, 7, 18, 0, 0),
    },
    {
      allDay: false,
      start: new Date(2017, 9, 7, 17, 0, 0),
      end: new Date(2017, 9, 7, 18, 30, 0),
    },
    {
      allDay: false,
      start: new Date(2017, 9, 7, 17, 0, 0),
      end: new Date(2017, 9, 7, 17, 30, 0),
    },
  ]
  const fevents = [
    {
      allDay: true,
      start: new Date(2017, 9, 7, 17, 0, 0),
      end: new Date(2017, 9, 8, 17, 30, 0),
    },
  ]
  const nodes = {
    0: {level: 0, children: [1, 2, 3], depth: 3},
    1: {level: 1, children: [], depth: 3},
    2: {level: 1, children: [3], depth: 3},
    3: {level: 2, children: [], depth: 3},
    4: {level: 0, children: [5, 6], depth: 2},
    5: {level: 1, children: [], depth: 2},
    6: {level: 1, children: [], depth: 2},
  }

  const ctx = {
    events,
    fevents,
    nodes,
    locale: frLocale,
    dateFormat: 'DD',
    hourFormat: 'HH',
    matrix: [],
    type: 'daily',
    endHour: 'PT22H',
    startHour: 'PT06H',
    rowHeight: 30,
  }
  test('render', () => {
    const tree = mount(
      <DailyCalendar start={new Date(2017, 9, 9, 0, 0, 0, 0)}>
        {({getColumnProps}) => <div {...getColumnProps()} />}
      </DailyCalendar>,
      {
        context: ctx,
      }
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
  test('next Day', () => {
    const tree = mount(
      <DailyCalendar start={new Date(2017, 9, 9)}>
        {({nextDay, dateLabel}) => <span onClick={nextDay}>{dateLabel('DD')}</span>}
      </DailyCalendar>,
      {
        context: ctx,
      }
    )
    expect(tree.find('span').html()).toBe('<span>09</span>')
    tree.find('span').simulate('click')
    expect(tree.find('span').html()).toBe('<span>10</span>')
  })
  test('prev Day', () => {
    const tree = mount(
      <DailyCalendar start={new Date(2017, 9, 9)}>
        {({prevDay, dateLabel}) => <span onClick={prevDay}>{dateLabel('DD')}</span>}
      </DailyCalendar>,
      {
        context: ctx,
      }
    )
    expect(tree.find('span').html()).toBe('<span>09</span>')
    tree.find('span').simulate('click')
    expect(tree.find('span').html()).toBe('<span>08</span>')
  })
  test('gotoToday Day', () => {
    const tree = mount(
      <DailyCalendar start={new Date(2017, 9, 9)}>
        {({gotoToday, prevDay, dateLabel}) => (
          <div>
            <button onClick={gotoToday}>Hlo</button>
            <span onClick={prevDay}>{dateLabel('DD')}</span>
          </div>
        )}
      </DailyCalendar>,
      {
        context: ctx,
      }
    )
    expect(tree.find('span').html()).toBe('<span>09</span>')
    tree.find('span').simulate('click')
    expect(tree.find('span').html()).toBe('<span>08</span>')
    tree.find('button').simulate('click')
    expect(tree.find('span').html()).toBe(`<span>18</span>`)
  })
})
