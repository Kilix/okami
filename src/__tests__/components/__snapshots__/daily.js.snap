// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyCalendar render 1`] = `
<BD
  start={2017-10-09T00:00:00.000Z}
>
  <DailyCalendar
    dateFormat="DD"
    endHour="PT22H"
    events={
      Array [
        Object {
          "allDay": false,
          "end": 2017-10-07T14:00:00.000Z,
          "start": 2017-10-07T03:00:00.000Z,
        },
        Object {
          "allDay": false,
          "end": 2017-10-07T07:00:00.000Z,
          "start": 2017-10-07T05:00:00.000Z,
        },
        Object {
          "allDay": false,
          "end": 2017-10-07T13:30:00.000Z,
          "start": 2017-10-07T12:30:00.000Z,
        },
        Object {
          "allDay": false,
          "end": 2017-10-07T14:30:00.000Z,
          "start": 2017-10-07T13:00:00.000Z,
        },
        Object {
          "allDay": false,
          "end": 2017-10-07T18:00:00.000Z,
          "start": 2017-10-07T16:30:00.000Z,
        },
        Object {
          "allDay": false,
          "end": 2017-10-07T18:30:00.000Z,
          "start": 2017-10-07T17:00:00.000Z,
        },
        Object {
          "allDay": false,
          "end": 2017-10-07T17:30:00.000Z,
          "start": 2017-10-07T17:00:00.000Z,
        },
      ]
    }
    fevents={
      Array [
        Object {
          "allDay": true,
          "end": 2017-10-08T17:30:00.000Z,
          "start": 2017-10-07T17:00:00.000Z,
        },
      ]
    }
    hourFormat="HH"
    locale={
      Object {
        "formatDistance": [Function],
        "formatLong": [Function],
        "formatRelative": [Function],
        "formatters": Object {
          "Do MMM": [Function],
          "Do MMMM": [Function],
        },
        "formattingTokensRegExp": /\\(\\\\\\[\\[\\^\\\\\\[\\]\\*\\\\\\]\\)\\|\\(\\\\\\\\\\)\\?\\(x\\|ss\\|s\\|mm\\|m\\|hh\\|h\\|do\\|dddd\\|ddd\\|dd\\|d\\|aa\\|a\\|ZZ\\|Z\\|YYYY\\|YY\\|X\\|Wo\\|WW\\|W\\|SSS\\|SS\\|S\\|Qo\\|Q\\|Mo\\|MMMM\\|MMM\\|MM\\|M\\|HH\\|H\\|GGGG\\|GG\\|E\\|Do MMMM\\|Do MMM\\|Do\\|DDDo\\|DDDD\\|DDD\\|DD\\|D\\|A\\|\\.\\)/g,
        "localize": Object {
          "month": [Function],
          "months": [Function],
          "ordinalNumber": [Function],
          "timeOfDay": [Function],
          "timesOfDay": [Function],
          "weekday": [Function],
          "weekdays": [Function],
        },
        "match": Object {
          "month": [Function],
          "months": [Function],
          "ordinalNumber": [Function],
          "ordinalNumbers": [Function],
          "timeOfDay": [Function],
          "timesOfDay": [Function],
          "weekday": [Function],
          "weekdays": [Function],
        },
        "options": Object {
          "firstWeekContainsDate": 4,
          "weekStartsOn": 1,
        },
      }
    }
    matrix={Array []}
    nodes={
      Object {
        "0": Object {
          "children": Array [
            1,
            2,
            3,
          ],
          "depth": 3,
          "level": 0,
        },
        "1": Object {
          "children": Array [],
          "depth": 3,
          "level": 1,
        },
        "2": Object {
          "children": Array [
            3,
          ],
          "depth": 3,
          "level": 1,
        },
        "3": Object {
          "children": Array [],
          "depth": 3,
          "level": 2,
        },
        "4": Object {
          "children": Array [
            5,
            6,
          ],
          "depth": 2,
          "level": 0,
        },
        "5": Object {
          "children": Array [],
          "depth": 2,
          "level": 1,
        },
        "6": Object {
          "children": Array [],
          "depth": 2,
          "level": 1,
        },
      }
    }
    offset={0}
    rowHeight={30}
    showNow={false}
    start={2017-10-09T00:00:00.000Z}
    startHour="PT06H"
    type="daily"
  >
    <div />
  </DailyCalendar>
</BD>
`;
